log4j-kafka/src/main/java/org/apache/logging/log4j/kafka/appender/KafkaAppender.java

```diff
@@ -136,6 +136,18 @@ public B setRetryCount(final String retryCount) {
         return new Builder<B>().asBuilder();
     }
 
+    private static final String[] KAFKA_CLIENT_PACKAGES = new String[] { "org.apache.kafka.common", "org.apache.kafka.clients" };
```

Comment:
If you use a static init block, you can sort the array and then bsearch() it in the  lookup method.

Reply:
It seems like we are checking the package prefix, not the exact match. Is there any way to do prefix match with binary search? (Or, using regex would be better?)

Reply:
You are correct, I forgot that this is just a prefix match, discard my last comment. I'll review in the morning unless someone else does first.

Reply:
Great. Happy holidays!! :smile: